Automattic/wp-calypso
# Themes Discovery: Add track events

Related to:
- https://github.com/Automattic/dotcom-forge/issues/4307

We need to add tracking capabilities to the Discovery carousels' cards and navigation component.

Note: After the PR gets merged we will need to add the events to Tracks following the PCYsg-mUj-p2, the issue hasn't been created yet since the new event names might change.

## Proposed Changes

* Refactored the tracking architecture outside of the ThemesSelection component.
* Refactored the ThemesSelection component to use the new Tracking mechanism.
* Unified the tracking event properties so that all events include the same properties.
* Reduced the complexity of the ThemesSelection component thanks to the new tracking mechanism.
* Added tracking capabilities to the Collection component cards and navigation.

## Testing Instructions



* Open the Calypso Live Link.
* Open Production on another tab.
* Click on a theme in the showcase (Non-collections)
* Compare the event across the proposal and production.
* The properties on the proposal should include all the production properties. In some cases, there will be new properties added so that all events contain the same properties. The filterString is replaced by three props, tier, filter and search.
* Now open the LoTS using the feature flag `?flags=themes/discovery-lots`
* When clicking on a card and on the style variation buttons events should be registered.
* When clicking on the `See all` button or navigation arrows, events should be registered.

## Pre-merge Checklist
